sound/soc/intel/boards/skl_hda_dsp_generic.c

```diff
@@ -146,7 +146,7 @@ static int skl_hda_audio_probe(struct platform_device *pdev)
 
 	INIT_LIST_HEAD(&ctx->hdmi_pcm_list);
 
-	mach = dev_get_drvdata(&pdev->dev);
+	mach = (&pdev->dev)->platform_data;//dev_get_drvdata(&pdev->dev);
```

Comment:
In sof, dev_get_platdata is called to get platform_data

Reply:
And if it is changed to use platform_data, SST driver also should be changed to store platform_data.
And it is also affect my PR. Is this your final solution?

Reply:
yes, I need to check on the SST side, but the idea is to align everyone on the same methodology for machine drivers.
What SOF does for other cases is irrelevant since it remains internal.

Reply:
@plbossart do we still need this change after the next commit that passes the correct machine while registering the machine driver?

Reply:
fixed now, please double-check